javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1500 @@
+   * values.
+   * <p>
+   * The two's complement representation of {@code this} will be at least
+   * {@code bitLength() + 1} bits long.
+   * <p>
+   * The value will fit into an {@code int} if {@code bitLength() < 32} or into
+   * a {@code long} if {@code bitLength() < 64}.
+   *
+   * @return the length of the minimal two's complement representation for
+   *         {@code this} without the sign bit.
+   */
+  def bitLength(): Int = BitLevel.bitLength(this)
+
+  /**
+   * Returns a new {@code BigInteger} which has the same binary representation
+   * as {@code this} but with the bit at position n cleared. The result is
+   * equivalent to {@code this & ~(2^n)}.
+   * <p>
+   * <b>Implementation Note:</b> Usage of this method is not recommended as the
+   * current implementation is not efficient.
+   *
+   * @param n position where the bit in {@code this} has to be cleared.
+   * @return {@code this & ~(2^n)}.
+   * @throws ArithmeticException if {@code n < 0}.
+   */
+  def clearBit(n: Int): BigInteger = {
+    if (testBit(n)) BitLevel.flipBit(this, n)
+    else this
+  }
+
+  /**
+   * Compares this {@code BigInteger} with {@code val}. Returns one of the three
+   * values 1, 0, or -1.
+   *
+   * @param val value to be compared with {@code this}.
+   * @return {@code 1} if {@code this > val}, {@code -1} if {@code this < val} ,
+   *         {@code 0} if {@code this == val}.
+   * @throws NullPointerException if {@code val == null}.
+   */
+  def compareTo(bi: BigInteger): Int = {
+    if (sign > bi.sign) {
+      GREATER
+    } else if (sign < bi.sign) {
+      LESS
+    } else if (numberLength > bi.numberLength) {
+      sign
+    } else if (numberLength < bi.numberLength) {
+      -bi.sign
+    } else {
+      // Equal sign and equal numberLength
+      sign * Elementary.compareArrays(digits, bi.digits, numberLength)
+    }
+  }
+
+  /**
+   * Returns a new {@code BigInteger} whose value is {@code this / divisor}.
+   *
+   * @param divisor value by which {@code this} is divided.
+   * @return {@code this / divisor}.
+   * @throws NullPointerException if {@code divisor == null}.
+   * @throws ArithmeticException if {@code divisor == 0}.
+   */
+  def divide(divisor: BigInteger): BigInteger = {
+    if (divisor.sign == 0) {
+      throw new ArithmeticException("BigInteger divide by zero")
+    }
+    val divisorSign = divisor.sign
+    if (divisor.isOne) {
+      if (divisor.sign > 0) this else this.negate()
+    } else {
+      val thisSign = sign
+      val thisLen = numberLength
+      val divisorLen = divisor.numberLength
+      if (thisLen + divisorLen == 2) {
+        var bi = (digits(0) & 0xFFFFFFFFL) / (divisor.digits(0) & 0xFFFFFFFFL)
+        if (thisSign != divisorSign) {
+          bi = -bi
+        }
+        valueOf(bi)
+      } else {
+        val cmp = if (thisLen != divisorLen) if (thisLen > divisorLen) 1 else -1
+                  else Elementary.compareArrays(digits, divisor.digits, thisLen)
```

Comment:
Do not nest `if`s without braces. Moreover, if the `else` falls on the second line, then put the `if` on the next line and indented already. So in this case:
```scala
val cmp = {
  if (thisLen != divisiorLen) {
    if (thisLen > divisorLen) 1
    else -1
  } else {
    Elementary.compareArrays(digits, divisor.digits, thisLen)
  }
}
```